webapp/index.js

```diff
@@ -0,0 +1,253 @@
+    );
+  });
+
+  // choose only top 10 files or nodes and total qty
+  const sortedNodesData = filesData.nodes.sort((a, b) => {
+    return (
+      b.data.reduce((acc, element) => {
+        return (acc += element[1]);
+      }, 0) -
+      a.data.reduce((acc, element) => {
+        return (acc += element[1]);
+      }, 0)
+    );
+  });
+  return {
+    files: sortedFilesData.slice(0, 11),
+    nodes: sortedNodesData.slice(0, 11)
+  };
+}
+
+// date and time format for title and tooltips
+const dateTimeOptions = {
+  year: "numeric",
+  month: "short",
+  day: "2-digit",
+  hour: "2-digit",
+  minute: "2-digit"
+};
+
+// default charts are for last 10 days with 144 minutes aggregate
+async function loadAndDraw({
+  from = new Date(new Date().setDate(new Date().getDate() - 10)),
+  to = new Date(),
+  aggregate = 144
+} = {}) {
+  const jsonData = await loadData({ to, from, aggregate });
+  const data = transformToFiles(jsonData);
+  drawChart(
+    data.files,
+    "top-files",
+    `Top 10 files downloaded from ${from.toLocaleDateString(
+      [],
+      dateTimeOptions
+    )}`
+  );
+  drawChart(
+    data.nodes,
+    "top-nodes",
+    `Top 10 nodes from ${from.toLocaleDateString([], dateTimeOptions)}`
+  );
+}
+
+anychart.onDocumentLoad(() => {
+  loadAndDraw();
+});
+
+const buttons = document.getElementsByClassName("period-button");
+for (let button of buttons) {
+  button.onclick = event => {
+    let args = {};
+    if (event.target.id === "1hour") {
+      args = {
+        from: new Date(new Date().setHours(new Date().getHours() - 1)),
+        aggregate: 1
+      };
+    } else if (event.target.id === "12hours") {
+      args = {
+        from: new Date(new Date().setHours(new Date().getHours() - 12)),
+        aggregate: 1
+      };
+    } else if (event.target.id === "24hours") {
+      args = {
+        from: new Date(new Date().setDate(new Date().getDate() - 1)),
+        aggregate: 1
+      };
+    } else if (event.target.id === "10days") {
+      args = {
+        from: new Date(new Date().setDate(new Date().getDate() - 10)),
+        aggregate: 144
+      };
+    }
+    loadAndDraw(args);
+  };
+}
+
+function drawChart(data, container, title) {
+  //clear the container
+  document.getElementById(container).innerHTML = `<h4>${title}</h4>`;
+  const chart = anychart.line();
+  const lines = data.map(el => {
+    const line = chart.spline(el.data);
+    if (el.file) {
+      line.name(el.file);
+    } else if (el.node) {
+      line.name(el.node);
+    }
+    return line;
+  });
+  // set the container element
+  chart.container(container);
+
+  // enable the legend
+  chart.legend(true);
+  // set the layout of the legend
+  chart.legend().itemsLayout("vertical-expandable");
+  // set the position of the legend
+  chart.legend().position("bottom");
+  // set the maximum height and width of the legend
+  chart.legend().height("220px");
+  chart.height("500px");
+  const margin = chart.legend().margin();
+  margin.top(10);
```

Comment:
`margin` used only one time I think it is not necessary to define var for it.

Looks better to me:
```js
chart.legend().margin().top(10)
```